http_proxy support for mount_gcsfuse

peter-fe opened the pull request:
Added http_proxy to the mount.gcsfuse helper as it was done previously for the cli part (v.0.29.0), Closes #388

peter-fe commented:
@googlebot I signed it!

peter-fe commented:
@bjornleffler, can you do a review?
Would be great, thx a lot!

bjornleffler commented:
Thanks for contributing this.

I would change:
- Stderr to Stdout. It's a notification, not an error.
- Remove empty line 240.

Questions:
- Have you tried this. Does it work as expected?
- GCS should use mostly https. How about we prefer https_proxy with a fallback to http_proxy?

Thanks,
Bjorn

peter-fe commented:
@bjornleffler 

Good point with the https preference, I added the fallback to gcsfuse as well as there was only http_proxy respected. (actually the part I've stolen the code snippet from).

Adjusted the output to stdout and moved the output to the main gcsfuse code (otherwhise you'll see it twice as gcsfuse and mount.gcsfuse are get called in series) and removed the output from mount.gcsfuse.

It's not 100% perfect as actually two binaries get called, but should be sufficient to indicate that the https_proxy/http_proxy was used as long as the proxy parts stays in sync in both places.

Sadly it seems that the build pipeline is broken since some time, some of the checks fail due internal errors (internal/gcsx).

kumowoon1025 commented on tools/mount_gcsfuse/main.go:
```diff
@@ -232,6 +232,16 @@ func run(args []string) (err error) {
 	cmd := exec.Command(gcsfusePath, gcsfuseArgs...)
 	cmd.Env = append(cmd.Env, fmt.Sprintf("PATH=%s", path.Dir(fusermountPath)))
+
+        // Pass through the https_proxy/http_proxy environment variable,
```
What happens if https_proxy is specified in both the env var and e.g. an entry in fstab?

peter-fe replied:
Do you mean something like this in fstab?

`my-bucket /my-mount-point gcsfuse rw,dir_mode=770,file_mode=660,https_proxy=http://my-proxy:8080`

kumowoon1025 replied:
That's what I meant, but I guess the same thing happens if it's defined in the shell environment and you also specify it as an option with the CLI tool, seems to take environment variable value.

peter-fe replied:
Actually, at least on my linux, when appended to the fstab it gets as -o https_proxy=... to fusermount and fusermount fails with an unkown parameter exception.

peter-fe commented:
@bjornleffler,

Can you give it another shot and maybe get the pipeline running again?

Br

bjornleffler commented:
Looks good. Thanks for the contribution.

I'll build a new release soon.

peter-fe commented:
@bjornleffler

Great, thank you!

peter-fe commented:
@bjornleffler

Would the new release also include the .rpm  again? That would be great, as it was missing in that last release.

Thx for keeping this project alive!

Br

peter-fe commented:
@bjornleffler 

Hey Bjorn, have you found some time to make a new release including this changes?
I'm mainly interested in the rpm & dedicated tag for the change.

Br
Peter

bjornleffler commented:
Oups, I totally dropped the ball on this.

We're preparing some major changes to GCS Fuse, so I'm going to cut a new version tomorrow. Let me make sure that I get the signed rpms out quickly this time.

bjornleffler commented:
Please have a look and see if release 0.30.0 works for you.

I have pushed the new release to the Google rpm repository.

peter-fe commented:
Thank you very much Bjorn!
Install and testing went well, perfekt!
